src/config.rs

```diff
@@ -237,6 +244,12 @@ impl Config {
             ),
             endpoint_addr: parse_addr(matches.value_of("endpoint-addr"))
                 .with_context(|| "Invalid endpoint address")?,
+            host_addr: match matches.value_of("host-addr") {
+                Some(host_addr) => {
+                    Some(parse_addr(Some(host_addr)).with_context(|| "Invalid host address")?)
```

Comment:
This should additionally check that the IP version of the host address matches the version of the endpoint address (IPv4/IPv6).